numcodecs/vlen.pyx

```diff
@@ -250,7 +250,10 @@ class VLenBytes(Codec):
             l = lengths[i]
             store_le32(<uint8_t*>data, l)
             data += 4
-            encv = PyBytes_AS_STRING(values[i])
+            b = values[i]
+            if b is None or b == 0:  # treat these as missing value, normalize
```

Comment:
Like on line 231? It seems like `l = 0` in this case, there is nothing to copy, and these operations are wasted. I might be wrong...

Reply:
If you don't pass a bytes object, you will hit this assert when calling PyBytes_AS_STRING: https://github.com/python/cpython/blob/40fff90ae3d46843bb9d27c6a53ef61c861a3bb4/Include/cpython/bytesobject.h#L21

it would be equivalent to skip the PyBytes_AS_STRING and memcpy entirely if l = 0. Shall I update the PR to do that instead?